clang/Cursor.d

```diff
@@ -105,7 +169,184 @@ struct Cursor
         return clang_isCursorDefinition(cast(CXCursor) cx) != 0;
     }
 
-    @property Cursor definition () const
+    bool isTranslationUnit() const
+    {
+        auto kind = clang_getCursorKind(cast(CXCursor) cx);
+        return clang_isTranslationUnit(kind) != 0;
+    }
+
+    private bool _isPredefined() const
+    {
+        immutable ubyte[string] predefined = [
```

Comment:
Fixed, now I consider all symbols that are defined in **empty** translation unit internal. Another way would be to check the symbol extent (SourceRange) predefined symbols doesn't seem to have a file path assigned.